client/lib/importer/actions.js

```diff
@@ -141,49 +148,66 @@ export const mapAuthor = ( importerId, sourceAuthor, targetAuthor ) =>
 
 export function resetImport( siteId, importerId ) {
 	// We are done with this import session, so lock it away
-	lockImport( importerId );
+	const lockImportAction = {
+		type: IMPORTS_IMPORT_LOCK,
+		importerId,
+	};
+	Dispatcher.handleViewAction( lockImportAction );
 
-	Dispatcher.handleViewAction( {
+	const resetImportAction = {
 		type: IMPORTS_IMPORT_RESET,
 		importerId,
 		siteId,
-	} );
+	};
+	Dispatcher.handleViewAction( resetImportAction );
 
 	apiStart();
 	wpcom
-		.updateImporter( siteId, expiryOrder( siteId, importerId ) )
-		.then( apiSuccess )
-		.then( fromApi )
-		.then( receiveImporterStatus )
+		.updateImporter( siteId, createExpiryOrder( siteId, importerId ) )
+		.then( ( data ) => {
```

Comment:
Yes, it will eventually happen after `apiSuccess` is removed and `fromApi` is moved inside `receiveImporterStatus`. Then the handler will become just:
```js
.then( receiveImporterStatus )
```